tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,190 @@
+
+# from ceph.utils import get_node_by_id
+from tests.rbd.rbd_peristent_writeback_cache import (
+    PersistentWriteAheadLog,
+    fio_ready,
+    get_entity_level,
+)
+from tests.rbd.rbd_utils import initial_rbd_config
+from tests.rbd.test_rbd_persistent_writeback_cache_flush import kill_fio
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def verify_cache(pwl, config, cache_client, image_spec):
+    """Method to verify rbd persistnat cache test.
+    Args:
+        config: test config
+        cache_client: client node
+        image_spec: image spec
+    """
+    log.info("Writing data to the image using FIO jobs and checking cache status")
+    with parallel() as p:
+        p.spawn(run_fio, **fio_ready(config, cache_client))
+        pwl.check_cache_file_exists(
+            image_spec,
+            config["fio"].get("runtime", 120),
+            **config,
+        )
+
+
+def run(ceph_cluster, **kw):
+    """RBD Persistent write back cache with cluster operations
+
+    Args:
+        ceph_cluster: ceph cluster object
+        **kw: test parameters
+    """
+    log.info(
+        "Running test - CEPH-83574891 - Verify ceph cluster operations \
+        with persistent cache enabled in SSD mode with IO in-progress"
+    )
+
+    config = kw.get("config")
+    rbd_obj = initial_rbd_config(**kw)["rbd_reppool"]
+    pool = config["rep_pool_config"]["pool"]
+    image_spec = (
+        f"{config['rep_pool_config']['pool']}/{config['rep_pool_config']['image']}"
+    )
+    # config["image_spec"] = image_spec
+
+    cache_client = ceph_cluster.get_nodes(role="client")[0]
+
+    admin_node = ceph_cluster.get_nodes(role="_admin")
+
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mon_obj = MonitorWorkflows(node=cephadm)
+
+    pwl = PersistentWriteAheadLog(rbd_obj, cache_client, config.get("drive"))
+    config_level, entity = get_entity_level(config)
+
+    try:
+        # Configute cache client
+        pwl.configure_cache_client()
+
+        # Configure PWL
+        pwl.configure_pwl_cache(
+            config["rbd_persistent_cache_mode"],
+            config_level,
+            entity,
+            config["cache_file_size"],
+        )
+
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        # Stop writing data to the image by killing the FIO process,
+        # then start FIO again and check cache status
+        log.info("Test to kill FIO process and restarting FIO to check cache status")
+        with parallel() as p:
+            p.spawn(run_fio, **fio_ready(config, cache_client))
+            p.spawn(kill_fio, pwl.rbd)
+
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        # Restart mon service along with cache test
+        log.info("Test to restarting mon service along with pwl cache test")
+        with parallel() as p:
+            p.spawn(rados_obj.restart_daemon_services, "mon")
```

Comment:
if restart_daemon_services is the single method running inside a parallel loop it doesn't make sense. Did you mean to run verify_cache also as a parallel method? If yes, then you should add p.spawn for calling verify_cache as well.

Reply:
done